webapp/src/pages/EditChatroomDetailsPage/EditChatroomDetailsPage.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  LargeAvatar,
+  Loading,
+  ToggleSwitch,
+} from '../../shared/components';
+import {
+  CHAT_URL,
+  EDIT_AVATAR_URL,
+  WILDCARD_AVATAR_URL,
+} from '../../shared/urls';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { useCallback, useEffect, useState } from 'react';
+import { ResponseError, UpdateChatroomDto } from '../../shared/generated';
+import './EditChatroomDetailsPage.css';
+import { chatApi } from '../../shared/services/ApiService';
+import { useData } from '../../shared/hooks/UseData';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+
+export default function CreateChatroomPage() {
+  const { chatroomId } = useParams();
+  const navigate = useNavigate();
+  const { goBack } = useNavigation();
+  const { warn, notify } = useNotificationContext();
+
+  const getChatRoomById = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId! }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading } = useData(getChatRoomById);
+  const [isPublic, setIsPublic] = useState(false);
+
+  const initialFormValues: UpdateChatroomDto = {
+    name: isLoading ? '' : chatroom!.name,
+    oldPassword: '',
+    password: '',
+    confirmationPassword: '',
+    isPublic: isPublic,
+  };
+  const [formValues, setFormValues] =
+    useState<UpdateChatroomDto>(initialFormValues);
+
+  useEffect(() => {
+    if (chatroom) {
+      setIsPublic(chatroom.isPublic);
+    }
+  }, [chatroom]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((previousValues: any) => {
+      return { ...previousValues, [name]: value };
+    });
+  };
+
+  function hasValidFormValues() {
+    if (formValues.password !== formValues.confirmationPassword) {
+      warn('Passwords are different');
+      return false;
+    }
+    return true;
+  }
+
+  async function updateChatroomDetails() {
+    if (!hasValidFormValues()) {
+      return;
+    }
+    try {
+      await chatApi.chatControllerUpdateChatroom({
+        chatroomId: chatroomId!,
+        updateChatroomDto: {
+          name: formValues.name ? formValues.name : initialFormValues.name,
+          oldPassword: formValues.oldPassword || null,
+          password: formValues.password || null,
+          confirmationPassword: formValues.confirmationPassword || null,
+          isPublic: isPublic,
+        },
+      });
+      notify('Chatroom details successfully updated');
+      navigate(CHAT_URL);
+    } catch (error) {
+      let errMessage = '';
+      if (error instanceof ResponseError) {
+        errMessage = `${error.response.statusText}`;
+      } else if (error instanceof Error) {
+        errMessage = `${error.message}`;
+      }
+      warn(errMessage);
```

Comment:
If we send a wrong `old password`  we see the `Forbidden` snackbar, but I'd expect to see the whole string from `transcendence-app/src/chat/chat.service.ts` L152 `      throw new ForbiddenException('Incorrect password');` So that the user attempting the change can understand what went wrong.

Also, I cannot try with a different, correct password anymore after I input an incorrect one. For some reason I can do it only if I reload.

Reply:
I am working on it, and on some improvements